src/drivers/azure-storage-blob.ts

```diff
@@ -24,41 +24,69 @@ export interface AzureStorageBlobOptions {
   accountKey?: string;
 
   /**
-   * The SAS key. If provided, the account key will be ignored.
+   * The SAS token. If provided, the account key will be ignored. Include at least read, list and write permissions to be able to list keys.
    */
   sasKey?: string;
 
+  /**
+   * The SAS URL. If provided, the account key, SAS key and container name will be ignored.
+   */
+  sasUrl?: string;
+
   /**
    * The connection string. If provided, the account key and SAS key will be ignored. Only available in Node.js runtime.
    */
   connectionString?: string;
+
+  /**
+   * Storage account endpoint suffix. Need to be changed for Microsoft Azure operated by 21Vianet, Azure Government or Azurite.
+   * @default ".blob.core.windows.net"
+   */
+  endpointSuffix?: string;
 }
 
 const DRIVER_NAME = "azure-storage-blob";
 
 export default defineDriver((opts: AzureStorageBlobOptions) => {
   let containerClient: ContainerClient;
+  const endpointSuffix = opts.endpointSuffix || ".blob.core.windows.net";
   const getContainerClient = () => {
     if (containerClient) {
       return containerClient;
     }
-    if (!opts.accountName) {
-      throw createError(DRIVER_NAME, "accountName");
+    if (!(opts.connectionString || opts.sasUrl) && !opts.accountName) {
```

Comment:
this negated logic reads strangely

`accountName` only becomes optional if you provide a `sasUrl` or a `connectionString` right?

this could be made clearer by something like this:

```ts
if (!opts.sasUrl && !opts.connectionString && !opts.accountName) {
```

Reply:
I have changed it